src/Mapping/ReflectionReadonlyProperty.php

```diff
@@ -42,7 +43,7 @@ public function setValue(mixed $objectOrValue, mixed $value = null): void
 
         assert(is_object($objectOrValue));
 
-        if (parent::getValue($objectOrValue) !== $value) {
+        if (serialize(parent::getValue($objectOrValue)) !== serialize($value)) {
```

Comment:
Another way to do it would be using a loose comparison, although I'm not sure if there are any performance implications. I'd like to know if you've considered it though.

https://github.com/doctrine/orm/blob/982d6060a3c71574a986da0f0e466669052791a5/src/UnitOfWork.php#L2319-L2320

Reply:
I didn't try it indeed. I'm not a big fan of loose comparison, which often leads to unexpected consequences in PHP.

Reply:
Ok, I'm undecided myself. Let's see what other maintainers have to say.

Reply:
this looks extremely expensive to call serialize

Reply:
> this looks extremely expensive to call serialize

"extremely", really?